server/sner/plugin/auror_hostnames/core.py

```diff
@@ -0,0 +1,167 @@
+"""Core functions for auror_hostnames plugin"""
+
+from socket import getaddrinfo
+import ipaddress
+import os
+import subprocess
+
+import dns.exception
+import dns.zone
+import dns.rdataclass
+import dns.rdatatype
+
+
+def get_repos(git_server, key_path) -> list:
+    """
+    Get DNS repositories from git server
+    """
+    cmd = ["ssh", "-i", key_path, f"git@{git_server}", "info"]
+    output = subprocess.check_output(cmd, text=True).splitlines()[2:]  # Skip the first line
+    repos = [line.split("\t")[-1] for line in output]
+
+    return repos
+
+
+def clone_dns_repo(git_server, repo, key):
+    """Clone DNS zones from git repository"""
+    env = os.environ.copy()
```

Comment:
would prefer to use deepcopy, not really issue in os.environ but be consistent with rest of the code